kolibri/plugins/management/assets/src/vue/modal/index.vue

```diff
@@ -40,27 +44,87 @@
 
 <script>
 
+  const vue = require('vue');
+
   module.exports = {
     props: {
-      btntext: {
+      title: {
         type: String,
-        default: 'Open Modal',
+        required: true,
+      },
+      invisibleTitle: {
+        type: Boolean,
+        default: false,
+      },
+      // Modal options
+      disableClose: {
+        type: Boolean,
+        default: false,
+        required: false,
+      },
+      backgroundClickClose: {
+        type: Boolean,
+        default: true,
+        required: false,
+      },
+      // useed to toggle error message in header
+      error: {
+        type: Boolean,
+        default: false,
+      },
+    },
+    ready() {
+      if (this.disableClose) {
+        this.$off('close');
+      }
+    },
+    events: {
+      open() {
```

Comment:
Like we discussed: I'm still iffy on using events like this because it feels un-fluxy. But we'll cycle back to it at a later point.